test/Microsoft.AspNetCore.Authentication.WsFederation.Test/WsFederationTest.cs

```diff
@@ -0,0 +1,293 @@
+            response = await httpClient.PostAsync(queryItems["wreply"], new FormUrlEncodedContent(kvps));
+
+            Assert.Equal(HttpStatusCode.Found, response.StatusCode);
+
+            var request = new HttpRequestMessage(HttpMethod.Get, response.Headers.Location);
+            var cookies = SetCookieHeaderValue.ParseList(response.Headers.GetValues(HeaderNames.SetCookie).ToList());
+            foreach (var cookie in cookies)
+            {
+                if (cookie.Value.HasValue)
+                {
+                    request.Headers.Add(HeaderNames.Cookie, new CookieHeaderValue(cookie.Name, cookie.Value).ToString());
+                }
+            }
+            response = await httpClient.SendAsync(request);
+
+            // Did the request end in the actual resource requested for
+            Assert.Equal(WsFederationDefaults.AuthenticationScheme, await response.Content.ReadAsStringAsync());
+        }
+
+        [Fact]
+        public async Task ValidUnsolicitedTokenIsAccepted()
+        {
+            var builder = new WebHostBuilder()
+                .ConfigureServices(ConfigureAppServices)
+                .Configure(ConfigureApp);
+            var server = new TestServer(builder);
+
+            var httpClient = server.CreateClient();
+
+            // Send an invalid token and verify that the token is not honored
+            var kvps = new List<KeyValuePair<string, string>>();
+            kvps.Add(new KeyValuePair<string, string>("wa", "wsignin1.0"));
+            kvps.Add(new KeyValuePair<string, string>("wresult", File.ReadAllText(@"ValidToken.xml")));
+            kvps.Add(new KeyValuePair<string, string>("suppressWctx", "true"));
+            var response = await httpClient.PostAsync(server.BaseAddress + "signin-wsfed", new FormUrlEncodedContent(kvps));
+
+            Assert.Equal(HttpStatusCode.Found, response.StatusCode);
+
+            var request = new HttpRequestMessage(HttpMethod.Get, response.Headers.Location);
+            var cookies = SetCookieHeaderValue.ParseList(response.Headers.GetValues(HeaderNames.SetCookie).ToList());
+            foreach (var cookie in cookies)
+            {
+                if (cookie.Value.HasValue)
+                {
+                    request.Headers.Add(HeaderNames.Cookie, new CookieHeaderValue(cookie.Name, cookie.Value).ToString());
+                }
+            }
+            response = await httpClient.SendAsync(request);
+
+            // Did the request end in the actual resource requested for
+            Assert.Equal(WsFederationDefaults.AuthenticationScheme, await response.Content.ReadAsStringAsync());
+        }
+
+        [Fact]
+        public async Task InvalidTokenIsRejected()
+        {
+            var builder = new WebHostBuilder()
+                .ConfigureServices(ConfigureAppServices)
+                .Configure(ConfigureApp);
+            var server = new TestServer(builder);
+
+            var httpClient = server.CreateClient();
+
```

Comment:
Can we move all the builder/server/client creation into a `CreateClient()`, the tests all seem to only use httpClient right?